operator/builtin/input/file/design.md

```diff
@@ -0,0 +1,146 @@
+# Fingerprints
+
+Files are identified and tracked using fingerprints. A fingerprint is the first `N` bytes of the file, with the default for `N` being `1000`. 
+
+### Fingerprint Growth
+
+When a file is smaller than `N` bytes, the fingerprint is the length of the file. A fingerprint that is less than `N` bytes will be compared to other fingerprints using a prefix check. As the file grows, its fingerprint will be updated, until it reaches the full size of `N`.
```

Comment:
>the fingerprint is the length of the file
Is fingerprint literally the length? Or the fingerprint is the entire file when the file length is < N?

Reply:
Good point. Will rephrase